### PR TITLE
Actually Fix the Bottom Bar Bug

### PR DESCRIPTION
**Pull request by evancrow**
Half-height `BottomBar` bug was caused by the safe-area value getting set high by the "Allow Paste" iOS 16 feature and not being reset on its dismissal. This bug is fixed by ignoring the safe area. 

I came across [this thread](https://stackoverflow.com/questions/63968241/swiftui-in-ios14-keyboard-avoidance-issues-and-ignoressafearea-modifier-issues) that had a method that worked for ignoring the keyboard `SafeArea`. (I had some prior issues where `ignoreSafeArea` wouldn't work on the `BottomBar`).

Also did some refactoring.

### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings

https://user-images.githubusercontent.com/37356984/195257811-33eca8b8-70c3-455c-826b-53d6797f8c0d.mp4

## Issues addressed
Fixed: #4379

### CODE REVIEW

**Comment by darinwf** [comment]
@evancrow - could you expand on how this fix works? (i know you explained it to me over Zoom, but would be good to have it written down for others to see too, thanks!)

File: Client/Frontend/Browser/BrowserBottomBarView.swift [hunk 1]
```diff
@@ -8,31 +8,25 @@ import SwiftUI
 
 struct BrowserBottomBarView: View {
     @EnvironmentObject var chromeModel: TabChromeModel
-    @EnvironmentObject var overlayManager: OverlayManager
```
Comment (darinwf) [inline Client/Frontend/Browser/BrowserBottomBarView.swift:11]: nice to eliminate a dependency

**Review by darinwf — APPROVED** [review]

**Comment by evancrow** [comment]
Updated my description.

**Comment by evancrow** [comment]
I win tests!!

**Pull request merged**